### DISCUSSION TITLE
Can I use state = AsyncData(state.value!.copyWith(newState)); to update state?

### DISCUSSION THREAD

**Discussion by devnta** [rrousselGit/riverpod#3474] [Q&A] [answered]
Is it safe when I use below code to update state?
```
state = AsyncData(state.value!.copyWith(newState));
```

**Comment by idispatch75** [comment, marked as answer]
What may not be safe?
If your concern is that `state.value` may be null, you can just use `state` methods to test that and update your state accordingly.
If you use this code in response to user interaction, then you know whether this interaction is possible only when the state is loaded, and in this case you can just use `state.requireValue` (this avoid the !).
Also inspect the API of AsyncValue, there are many interesting methods in there.
